file_paths.md

```diff
@@ -0,0 +1,108 @@
+
+    /etc/puppetlabs/code                  # :codedir                     contents moved from :confdir
+        environments                      # :environmentpath
+        hiera.yaml                        # :hiera_config
+        hieradata                         # n/a
+        modules                           # user modulepath
+
+    /etc/puppetlabs/mcollective
+        client.cfg
+        facts.yaml
+        server.cfg
+
+    /etc/puppetlabs/puppet                # :confdir                     /etc/puppet
+        auth.conf                         # :rest_authconfig
+        autosign.conf                     # :autosign
+        binder_config.yaml                # :binder_config
+        csr_attributes.yaml               # :csr_attributes
+        custom_trusted_oid_mapping.yaml   # :trusted_oid_mapping_file
+        device.conf                       # :deviceconfig
+        fileserver.conf                   # :fileserverconfig
+        puppet.conf                       # :config
+        routes.yaml                       # :route_file
+        ssl                               # :ssldir                      /etc/puppet/ssl
+
+    /opt/puppetlabs/bin                   # symlink targets of puppet related binaries, excluding ruby, openssl
+        cfacter@                          -> /opt/puppetlabs/puppet/bin/cfacter
+        facter@                           -> /opt/puppetlabs/puppet/bin/facter
+        hiera@                            -> /opt/puppetlabs/puppet/bin/hiera
+        mco@                              -> /opt/puppetlabs/puppet/bin/mco
+        mcollectived@                     -> /opt/puppetlabs/puppet/bin/mco
+        puppet@                           -> /opt/puppetlabs/puppet/bin/puppet
+
+    /opt/puppetlabs/mcollective/
+        plugins
+
+    /opt/puppetlabs/puppet                # ruby-puppet root
+        bin
+            cfacter
+            facter
+            gem
+            hiera
+            mco
+            mcollectived
+            openssl
+            puppet
+            ruby
+        cache                             # :vardir                      /var/lib/puppet
+            bucket                        # :bucketdir
+            client_yaml                   # :clientyamldir
+            client_data                   # :client_datadir
+            clientbucket                  # :clientbucketdir
+            devicedir                     # :devices
+            facts.d                       # :pluginfactdest
+            lib                           # :libdir
+            facts                         # used to generate :factpath
+            puppet-module                 # :module_working_dir
+            reports                       # :reportdir
+            server_datadir                # :server_data
+            state                         # :statedir
+            yaml                          # :yamldir
+        include
+        lib
```

Comment:
Using puppet-agent 3a63b02c9d471c7c80b74ddd547539c3812f58f9 (from a day or so ago) on rhel7, I only see static lib, not an `.so`. Will this be changing?

```
# ls -la /opt/puppetlabs/puppet/lib/*ruby*
-rw-r--r--. 1 root root 5203372 Feb 15 19:31 /opt/puppetlabs/puppet/lib/libruby-static.a
```